utils_linux.go

```diff
@@ -246,7 +246,7 @@ func (r *runner) run(config *specs.Process) (int, error) {
 	// Setting up IO is a two stage process. We need to modify process to deal
 	// with detaching containers, and then we get a tty after the container has
 	// started.
-	handler := newSignalHandler(r.enableSubreaper, r.notifySocket)
+	handlerCh := newSignalHandler(r.enableSubreaper, r.notifySocket)
 	tty, err := setupIO(process, r.container, config.Terminal, detach, r.consoleSocket)
```

Comment:
Did you try to see if doing `setupIO` and `setupPidfdSocket` in parallel also gave some speedup? I'm trying to understand why `signal.Notify` is so expensive, or if running it parallel let some other code unblocked something else that causes the noticeable speedup...